cpp/include/cugraph_c/graph_functions.h

```diff
@@ -184,6 +184,14 @@ cugraph_error_code_t cugraph_extract_induced_subgraph(
   cugraph_induced_subgraph_result_t** result,
   cugraph_error_t** error);
 
+cugraph_error_code_t cugraph_replicate_edgelist(
```

Comment:
Need a documentation block here.

We probably should add support for edge type and edge id as well, to be consistent with the rest of the graph creation API.  API changes in the C API are harder (there are no optional parameters, so deprecating a change is complex).

Reply:
Also, we're technically doing an `allgather` in the implementation, so it might be worth calling this function `cugraph_allgather_edgelist`.  If the user were to call this with a distributed edge list then `allgather` is the right name.  We can describe in the documentation that a user can *broadcast* from a single node by passing in empty lists from every rank that doesn't have data.

Reply:
I added documentation and updated the function name to `cugraph_allgather_edgelist`. I also added support for edge_ids and edge_type_ids however, I think we should rename `cugraph_induced_subgraph_result_t` later because it is quite odd to have it as a result type when doing an `allgatherv`.

Reply:
Agree this needs a new name.  I don't want to create different objects, that would be overkill.  Finding a better name that can be used by this logic, the induced subgraph logic, egonet, ktruss and anything else returning a list of edges would be ideal.  I'm inclined to do that in a separate PR as it affects other algorithms and is orthogonal to this.  Perhaps add a FIXME for now.

Reply:
I added already added a `FIXME` in L205